throttler/throttler.go

```diff
@@ -0,0 +1,78 @@
+package throttler
+
+import (
+	"context"
+	"fmt"
+	"sync/atomic"
+	"time"
+)
+
+// ThrottledAction is a common interface for actions that can be executed
+type ThrottledAction interface {
+	Execute()
```

Comment:
You don't need this abstraction if we use  a `runner` which already receives a context in its `Run` method.